### PR TITLE
Add CIO hanging test.

### PR DESCRIPTION
**Pull request by dmitrievanthony**
**Subsystem**
CIO Client

**Motivation**
We need to have such a test to fix issue #1720.

### CODE REVIEW

File: ktor-client/ktor-client-cio/jvm/test/io/ktor/client/engine/cio/CIORequestTest.kt [hunk 1]
```diff
@@ -45,4 +46,30 @@ class CIORequestTest : TestWithKtor() {
...
+            }
+        }
+
+        test { client ->
+            withTimeout(5000) {
+                for (i in 0..1000) {
+                    try {
+                        client.get<String>("http://something.wrong")
+                    } catch (e: Throwable) {
+                        if (e is CancellationException) {
```
Comment (e5l) [inline ktor-client/ktor-client-cio/jvm/test/io/ktor/client/engine/cio/CIORequestTest.kt:66]: Codestyle

Comment (dmitrievanthony) [reply ktor-client/ktor-client-cio/jvm/test/io/ktor/client/engine/cio/CIORequestTest.kt]: Fixed.

**Comment by ssp** [comment]
Any success analysing the issue yet?

**Comment by dmitrievanthony** [comment]
> Great that you were able to reproduce it in such a straightforward way!
> Any success analysing the issue yet?

As far as I understand @e5l has fixed this issue in his branch that is not merged into master yet. This test should validate the fix after the merge. Hi @e5l, do correct me if I'm wrong.

**Comment by e5l** [comment]
obsolete

**Pull request closed without merging**